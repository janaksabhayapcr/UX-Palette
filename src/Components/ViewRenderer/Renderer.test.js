import React, { Component } from 'react';
import Renderer from './Renderer';
import { configure, shallow, wrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

// jest mock functions (mocks this.props.func)
const addLogToQueue = jest.fn();
const updateNumberOfViewsLoaded = jest.fn();
const deleteView = jest.fn();
const onClick = jest.fn();
const onChange = jest.fn();

// defining this.props
const baseProps = {
  services: {
    Dashboard: {
      URL: '',
      subscription_key: {},
    },
    Renderer: {
      URL: '',
    },
  },

  layoutViewId: {},
  drilldownRender: {},
  viewId: {},
  logBody: {},
  renderId: {},
  module: {
    moduleId: {},
  },
  viewXML: {},
  context: [
    {
      Parameter: [],
      Value: [],
    },
  ],
  addLogToQueue,
  updateNumberOfViewsLoaded,
  transforms: {
    normalizeTransform: {},
    renderSchemaTransform: {},
    shapeTransform: {},
    filterTransform: {},
    aggregateTransform: {},
    pivotStagingTransform: {},
    pivotTransform: {},
    sortTransform: {},
    finalcolumnTransform: {},
    styleSheetTransform: {},
  },
  chartConfiguration: {},
  chartSeries: {},
  chartFinal: {},
  deleteView,
  renderLocation: {},
  user: {},
  firmContext: {},
  editMode: {},
  getCookie: {},
  contextParameters: {},
  liquidFunctions: {},
  liquidTemplates: {},
  systemLayout: {},
  userRoles: {},
  allowView: {},
  allowEdit: {},
  sideToRender: {},
  onClick,
  onChange,
};
describe('View Renderer : RendererTest', () => {
  let wrapper;
  let tree;

  beforeEach(() => (wrapper = shallow(<Renderer {...baseProps} />)));

  it(' Should render with all of the props', () => {
    tree = renderer.create(<Renderer {...baseProps} />);
    let treeJson = tree.toJSON();
    expect(treeJson).toMatchSnapshot();
    tree.unmount();
  });

  it('renders a modal portal', () => {
    const isOpen = wrapper.state('isOpen');
    const modalPortal = wrapper.find('#modal-container');
    expect(isOpen).toBeTruthy;
    expect(modalPortal).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
