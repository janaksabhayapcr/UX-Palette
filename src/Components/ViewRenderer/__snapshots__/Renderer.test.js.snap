// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Renderer : RendererTest  Should render with all of the props 1`] = `<div />`;

exports[`View Renderer : RendererTest renders a modal portal 1`] = `
<div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    isOpen={false}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "bottom": "auto",
          "height": "25%",
          "left": "50%",
          "marginRight": "-50%",
          "overflowY": "auto",
          "right": "auto",
          "top": "25%",
          "updateViewXMLValue": "translate(-50%, -50%)",
          "width": "35%",
        },
      }
    }
  >
    <div
      id="modal-container"
    >
      <ViewOverride
        context={
          Array [
            Object {
              "Parameter": Array [],
              "Value": Array [],
            },
          ]
        }
        extractDataFromXML={[Function]}
        settingModalOpen={false}
        user={Object {}}
      />
      <div
        id="modal-button-container"
      >
        <button
          className="modal-button"
          id="modal-btn-cancel"
          onClick={[Function]}
        >
          Cancel
        </button>
      </div>
    </div>
  </Modal>
</div>
`;
